index.html

```diff
@@ -7176,6 +7414,334 @@ <h2 id="MediaStreamAudioSourceNode">
+          in parallel</a>, as described in [[HTML]].
+        </p>
+        <p>
+          The synchronous section is executed on the <a>control thread</a>, and
+          happens immediately. If it fails, the method execution is aborted,
+          possibly throwing an exception. If it succeeds, a <a>control
+          message</a>, encoding the operation to be executed on the
+          <a>rendering thread</a> is enqueued on the <a>control message
+          queue</a> of this <a>rendering thread</a>.
+        </p>
+        <p>
+          The synchronous and asynchronous sections order with respect to other
+          events MUST be the same: given two operation <em>A</em> and
+          <em>B</em> with respectively synchronous and asynchronous section
+          <em>A<sub>Sync</sub></em> and <em>A<sub>Async</sub></em>, and
+          <em>B<sub>Sync</sub></em> and <em>B<sub>Async</sub></em>, if
+          <em>A</em> happens before <em>B</em>, then <em>A<sub>Sync</sub></em>
+          happens before <em>B<sub>Sync</sub></em>, and
+          <em>A<sub>Async</sub></em> happens before <em>B<sub>Async</sub></em>.
+        </p>
+      </section>
+      <section id="rendering-loop">
+        <h3>
+          Rendering an audio graph
+        </h3>
+        <p>
+          Rendering an audio graph is done in blocks of 128 samples-frames.
+        </p>
+        <p>
+          Operations that happen <dfn data-lt="atomic">atomically</dfn> on a
+          given thread can only be executed when no other <a>atomic</a>
+          operation is running on another thread.
+        </p>
+        <p>
+          The algorithm for rendering a block of audio from an
+          <a>AudioContext</a> <em>G</em> with an <a href=
+          "#control-message-queue">control message queue</a> <em>Q</em> is as
+          follows.
+        </p>
+        <ol>
+          <li>Let <em>Q<sub>rendering</sub></em> be an empty <a>control message
+          queue</a>. <a>Atomically</a> <a>swap</a>
+          <em>Q<sub>rendering</sub></em> and <em>Q</em>
+          </li>
+          <li>Until there are no more messages in
+          <em>Q<sub>rendering</sub></em>, execute the following steps:
+            <ol>
+              <li>Execute the asynchronous section of the <a>oldest message</a>
+              of <em>Q<sub>rendering</sub></em>.
+              </li>
+              <li>Remove the <a>oldest message</a> of
+              <em>Q<sub>rendering</sub></em>.
+              </li>
+            </ol>
+          </li>
+          <li>Order the <a>AudioNode</a>s of the <a>AudioContext</a> to be
+          processed.
+            <ol>
+              <li>Let <em>ordered</em> be an empty list of <a>AudioNode</a>s.
+              It will contain an ordered list of <a>AudioNode</a>s when this
+              ordering algorithm terminates.
+              </li>
+              <li>Let <em>nodes</em> be the set of all nodes belonging to an
```

Comment:
You haven't defined what this set really means.  Is it the set of all nodes created on the AudioContext?